cangen.c

```diff
@@ -722,96 +838,106 @@ int main(int argc, char **argv)
 		return 1;
 
 	while (running) {
-		frame.flags = 0;
+		cu.fd.flags = 0;
 
 		if (count && (--count == 0))
 			running = 0;
 
-		if (canfd) {
+		if (canxl) {
+			mtu = CANXL_MTU;
+			maxdlen = CANFD_MAX_DLEN; /* generate up to 64 byte */
```

Comment:
This had two reasons:

1. reuse of the length generation from CAN FD frames
2. `cangen` is about "just generating traffic" and if you run a `candump` in another terminal and get a frame output of 2048 bytes this simply pollutes your screen.

Therefore I limited the CAN XL data length for `cangen` to 64 byte by default.
Of course this can be extended or we can make another option to remove that limit.
Would you think this makes sense?